validator/sawtooth_validator/metrics/metrics.py

```diff
@@ -0,0 +1,196 @@
+        )
+
+    def gauge(self, identifier, level, instance=None, tags=None):
+        if self._registry is None or self._disabled(identifier, level):
+            return self._mock_registry.gauge(identifier)
+
+        return self._registry.gauge(
+            self._join(identifier, instance, tags))
+
+    def counter(self, identifier, level, instance=None, tags=None):
+        if self._registry is None or self._disabled(identifier, level):
+            return self._mock_registry.counter(identifier)
+
+        return self._registry.counter(
+            self._join(identifier, instance, tags))
+
+    def timer(self, identifier, level, instance=None, tags=None):
+        if self._registry is None or self._disabled(identifier, level):
+            return self._mock_registry.timer(identifier)
+
+        return self._registry.timer(
+            self._join(identifier, instance, tags))
+
+    # Private methods
+    def _disabled(self, identifier, level):
+        """Check if the metric is enabled based on the config."""
+        return level < self._config.reporting_level
+
+    def _join(self, identifier, instance=None, tags=None):
+        """
+        Join the identifier tuple with periods ".", combine the arbitrary tags
+        with the base tags and the identifier tag, convert tags to "tag=value"
+        format, and then join everything with ",".
+        """
+        tag_list = []
+        if tags is not None:
+            tag_list.extend(tags.items())
+        tag_list.extend(self._base_tags)
+        if instance is not None:
+            tag_list.append(("instance", id(instance)))
+        return ".".join(identifier) + "," + ",".join(
+            "{}={}".format(k, v)
+            for k, v in tag_list
+        )
+
+
+class MetricsCollectorHandle:
+    def __init__(self, module_name):
+        self._module_name = module_name
+
+    def gauge(self, metric_name, level=DEFAULT, instance=None, tags=None):
+        return MetricsCollector.get_instance().gauge(
+            identifier=self._create_identifier(metric_name, instance),
+            level=level,
+            instance=instance,
+            tags=tags)
+
+    def counter(self, metric_name, level=DEFAULT, instance=None, tags=None):
+        return MetricsCollector.get_instance().counter(
+            identifier=self._create_identifier(metric_name, instance),
+            level=level,
+            instance=instance,
+            tags=tags)
+
+    def timer(self, metric_name, level=DEFAULT, instance=None, tags=None):
+        return MetricsCollector.get_instance().timer(
+            identifier=self._create_identifier(metric_name, instance),
+            level=level,
+            instance=instance,
+            tags=tags)
+
+    def _create_identifier(self, metric_name, instance=None):
+        if instance is None:
+            return (self._module_name, metric_name)
+        return (self._module_name, instance.__class__.__name__, metric_name)
+
+
+class MockMetricsRegistry:
```

Comment:
I'm not sure I'd call this "Mock", as that has test connotations.  I would consider naming these as "NoOp"

Reply:
NoOp seems fine. Originally I had Fake which was even worse.